### PR TITLE
Add "Just the HM Docs" Jekyll theme

### PR DESCRIPTION
**Pull request by joeleenk**
**PR includes**
- HM-ifies the ["Just the Docs" Jekyll theme for documentation](https://github.com/just-the-docs/just-the-docs)
- Includes some page and post style sample content to make it easy to get started

**Future work (for another PR)**
- Figure out how to easily add this to new and existing HM projects
- Style the docs to match our [HM design](https://www.figma.com/file/m1r8StcFu76qTStAvwzVXT/Human-Made?node-id=1506%3A2689&t=htfprBdy2cyEIAF5-0)

**Related Ticket** 
https://github.com/humanmade/commercial/issues/1171
https://github.com/humanmade/agency/issues/1352#issuecomment-1370143896

### CODE REVIEW

**Comment by alwaysblank** [comment]
I've changed the base for this PR to make it easier to review, but also to indicate merge order: #3 should be merged first, and when it is this PR will automatically update to `main`.

File: .gitignore
Comment (alwaysblank) [inline .gitignore]: Is this from a boilerplate gitignore, or is there an actual `vendor` directory generated that we want to ignore? Usually I see this in a PHP context, but this project doesn't include PHP (and likely never will).
```suggestion
```

File: CHANGELOG.md
Comment (alwaysblank) [inline CHANGELOG.md]: (Opinion alert) I don't feel like markdown files should contain template code (except in the case where it only serves to enhance visual presentation). _In general_ a markdown file is intended to be human-readable as-is—the markup is there only to enhance the visual presentation. I feel like this is doubly true for a file like the changelog, which is (IMO) meant to be a record of changes that's as easily accessible as possible. This approach means the changelog is _only_ usable on the rendered site; someone just downloading or visiting the repo can't use it.

I'd recommend taking another approach here, maybe one of the following:
- `CHANGELOG.md` is just a list of changes; `_changelog` is removed.
- `CHANGELOG.md` has text directing a user to go to `_changelog` to see a list of changes, and there is a proper template file to list out the changelog entries.
- Implement a separate build process to create an updated `CHANGELOG.md` with the entries in `_changelog` which releases are pushed.

File: README.md
Comment (alwaysblank) [inline README.md]:
```suggestion
- [announcements](https://humanmade.github.io/just-the-hm-docs/announcements.md)
- an [introduction](https://humanmade.github.io/just-the-hm-docs/introduction.md) page
- a [feature](https://humanmade.github.io/just-the-hm-docs/features.md) list
- [guides](https://humanmade.github.io/just-the-hm-docs/guides.md)
- [modules](https://humanmade.github.io/just-the-hm-docs/modules.md)
- and a [changelog](https://humanmade.github.io/just-the-hm-docs/changelog.md)
```

File: README.md
Comment (alwaysblank) [inline README.md]:
```suggestion
- automatic [navigation structure](https://humanmade.github.io/just-the-hm-docs/docs/navigation-structure)
- instant, full-text [search](https://humanmade.github.io/just-the-hm-docs/docs/search) and page indexing
- and a set of [UI components](https://humanmade.github.io/just-the-hm-docs/docs/ui-components) and authoring [utilities](https://humanmade.github.io/just-the-hm-docs/docs/utilities)
```

File: _announcements/post-0.md
Comment (alwaysblank) [inline _announcements/post-0.md]: This could be simplified a big by removing the `date` frontmatter and putting that into the filename, i.e. `_announcements/2023-02-01-breaking-change.md`. Jekyll supports parsing dates from the filename:
- https://jekyllrb.com/docs/posts/
- https://jekyllrb.com/docs/front-matter/#predefined-variables-for-posts

File: _layouts/announcement.html
Comment (alwaysblank) [inline _layouts/announcement.html]: Nitpick: For short things (i.e the announcements in this repo) this says `0 min` which is a weird concept (I read this instantly!). It would be nice to have a fallback to deal with <1 results.

File: announcements.md
Comment (alwaysblank) [inline announcements.md]: See note above regarding using markdown files in this way.

File: changelog.md
Comment (alwaysblank) [inline changelog.md]: See note above about using markdown files in this way.

Also, either this file or `CHANGELOG.md` should be removed, as they duplicate one another and result in two entries in the menu.

File: features.md
Comment (alwaysblank) [inline features.md]: See note above about using markdown files in this way.

File: index.md
Comment (alwaysblank) [inline index.md]:
```suggestion
- [announcements](announcements)
- an [introduction](introduction) page
- a [feature](features) list
- [guides](guides)
- [modules](modules)
- and a [changelog](changelog)
```

File: just-the-hm-docs.gemspec
Comment (alwaysblank) [inline just-the-hm-docs.gemspec]: This should be updated to include any additional templates.

**Review by alwaysblank — DISMISSED** [review]

File: .gitignore
Comment (joeleenk) [reply .gitignore]: It was boilerplate. I think it's safe to remove.

File: CHANGELOG.md
Comment (kadamwhite) [reply CHANGELOG.md]: I'd counter on two points: one, if we're tolerating frontmatter (which we must), I feel that the line between "content" and "template" blurs in a jekyll theme and don't personally agree that there's a strong need to separate them, given the purpose of this repository. My second argument, which I'd argue is the more salient one, is that separating it out the way you're suggesting feels more complicated and would increase friction in using this theme. Anything which requires introducing a separate build process, I'd recommend we hold off on until we are certain we'd need it.

@joeleenk thoughts?

Comment (joeleenk) [reply CHANGELOG.md]: I guess my question is whether we prefer to:

1. pull in smaller parts, like an archive page with posts, which requires a template of some kind, or
2. have larger, complete markdown files that are fully human readable

I personally prefer smaller, well organized bits of content most of the time, but I'm not sure if any of our projects will have so much content that it matters. 

I guess we could keep one example of how to use the templates, and switch the changelog to be static.

Comment (alwaysblank) [reply CHANGELOG.md]: > My second argument, which I'd argue is the more salient one, is that separating it out the way you're suggesting feels more complicated and would increase friction in using this theme.

I agree this is not optimal, which is why it was my last suggestion. I'd prefer the first option I suggested--one single human-readable `CHANGELOG.md`. At least personally, I don't feel this makes the theme meaningfully more difficult to use or update--if anything it makes it easier because you only need to edit a single file vs create a new file, make sure it has appropriate naming, frontmatter, etc.

> if we're tolerating frontmatter (which we must), I feel that the line between "content" and "template" blurs in a jekyll theme

I'm not sure I understand the argument here. Frontmatter isn't "meant for human consumption," sure, but it's metadata that describes the content that follows it: While it might not be easy for a human to read, it does _convey information_, which IMO makes it meaningfully different from liquid markup, which conveys no meaningful or useful information to a human. A markdown file containing only liquid markup is not helpful to a person looking for documentation in the repo; it's an impediment.

> I personally prefer smaller, well organized bits of content most of the time

Conceptually I do too--it just feels nicer--but in this case I feel like it complicates the system for updating the docs which is going to make updates less likely, and I'm not sure what additional benefit it provides beyond making sense organizationally. Without a larger system (definitely beyond the scope of this) that handles the UX/UI of creating and managing these files, I'd rather structure docs in the simplest (i.e. fewest files) way possible.

IMO one of the primary goals for documentation is that it should be as easy to create, update, and access as we can possibly make it, which is why I'm skeptical of splitting these files up and using `.md` files as templates. It is nice to have a site with the docs, but it's _also_ very valuable to have easy-to-read docs available right in the repo (either cloned to your local or on github). What is the benefit to the maintainer and the consumer in making it more difficult/time consuming to get to the actual changelog?

Comment (kadamwhite) [reply CHANGELOG.md]: > What is the benefit to the maintainer and the consumer in making it more difficult/time consuming to get to the actual changelog?

That's fair. I think I misunderstood the discussion due to a too-hasty review; I think trying to be less 'clever' in the theme, in favor of having the basic process of "take these markdown files and make them into a nice site, with as minimal an install process as possible," is in line with the original goals we had for the initiative.

File: README.md
Comment (kadamwhite) [inline README.md:13]: Feels like this reflects the classroom theme repo process -- didn't we decide to move towards using this more as a [remote_theme](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/adding-a-theme-to-your-github-pages-site-using-jekyll), so that you could add a config pointing to this repo to a given project's `docs/_config.yml` and GH would pick it up there? (assuming `docs/` is the publishing source, could also of course be GH pages).

may not be necessary to change at this time, but want to revisit it once we figure out the lowest-friction way to deploy this

File: guides/prerequisites.md
Comment (kadamwhite) [inline guides/prerequisites.md]: Nitpick, should this be a subsection of Installation? I'm not entirely sure how these pages would appear (if at all) in a site that consumes this theme, but I was a bit surprised to see this as its own page.

**Review by kadamwhite — APPROVED** [review]
Some of this we may want to iterate on, but here's a ✅ from me. It's looking good and I'm curious to get it merged and see what it looks like put all together on a target repo's site.

**Comment by kadamwhite** [comment]
It also looks like if you use this branch as-is for a `remote_theme`, the traditional GH pages deploy step breaks because it doesn't expect Sass:

```
  Rendering Markup: assets/css/just-the-docs-dark.scss
github-pages 227 | Error:  Invalid CSS after "  $rem-size: math": expected "}", was ".div($size, $ro..." on line 4
      Remote Theme: Cleaning up /tmp/jekyll-remote-theme-20230131-7-1bhucm8
```

I'm not entirely clear what the "build process" is for the parent theme we're forking, here

File: README.md
Comment (joeleenk) [reply README.md:13]: Yeah, I want to re-write the getting started docs, but I'm not clear enough on what the steps are yet.

**Comment by joeleenk** [comment]
> It also looks like if you use this branch as-is for a `remote_theme`, the traditional GH pages deploy step breaks because it doesn't expect Sass

I suspect it's a versioning thing, rather than sass itself. I updated this function that it's complaining about to use `math.div` instead of `/`. It works locally, but maybe the remote build process is using an old version of `node-sass`?

**Pull request merged**
